### PR TITLE
Feat: typed host-set form components - edit and view

### PR DESCRIPTION
**Pull request by DhariniJeeva**
this PR splits host-set into three types and a proxy form component similar to host catalog forms. 

TODO: added a placeholder for `filter` field in AWS and Azure. will be updating those fields when the modeling part is ready. 

[static](https://boundary-ui-jyybbvga2-hashicorp.vercel.app/scopes/s_xk1i4dk50v2/host-catalogs/host-catalog-id-0/host-sets)
[azure](https://boundary-ui-jyybbvga2-hashicorp.vercel.app/scopes/s_xk1i4dk50v2/host-catalogs/host-catalog-id-3/host-sets)
[aws](https://boundary-ui-jyybbvga2-hashicorp.vercel.app/scopes/s_xk1i4dk50v2/host-catalogs/host-catalog-id-5/host-sets)

### CODE REVIEW

**Review by randallmorey — APPROVED** [review]

**Comment by DhariniJeeva** [comment]
some of the provider translations will be consolidated and addressed in a separate PR

File: addons/core/translations/resources/en-us.yaml [hunk 1]
```diff
@@ -129,10 +129,15 @@ host-set:
...
   types:
-    static: Static
-    unknown: Unknown
-    aws: AWS
-    azure: Azure
+    static:
+      label: Static
+      help: Provide a list of hosts and host sets with static addresses
+    unknown:
+      label: Unknown
```
Comment (calcaide) [inline addons/core/translations/resources/en-us.yaml:136]: Do we have not `help` for unknown, aws and azure?

**Review by calcaide — APPROVED** [review]
LGTM! Good work!

File: addons/core/translations/resources/en-us.yaml (hunk 1 above)
Comment (DhariniJeeva) [reply addons/core/translations/resources/en-us.yaml:136]: for host-sets, I don't think we do but I will check this further and address it if we need it!

**Pull request merged**